cetmix_tower_server/tests/test_plan.py

```diff
@@ -1261,3 +1261,425 @@ def test_plan_with_another_plan_with_all_not_executable_condition(self):
+        self.write_and_invalidate(
+            self.plan_2, **{"server_ids": [(6, 0, [self.server_test_1.id])]}
+        )
+        # Ensure that managers cannot delete plan line actions assigned to servers to
+        # which they aren't subscribed
+        with self.assertRaises(AccessError):
+            plan_line_action_as_bob.unlink()
+        # Check that managers can't access  plan line actions assigned to servers
+        #  to which they aren't subscribed
+        with self.assertRaises(AccessError):
+            plan_line_action_read_result = plan_line_action_as_bob.read([])
+        # Check that managers have access to plan line assigned to servers
+        #  to which they are subscribed
+        self.server_test_1.message_subscribe([self.user_bob.partner_id.id])
+        plan_line_action_read_result = plan_line_action_as_bob.read([])
+        self.assertEqual(
+            plan_line_action_read_result[0]["name"],
+            plan_line_action_as_bob.name,
+            msg="Name should be the same",
+        )
+        # Check that users have access to plan line actions assigned to servers
+        #  to which they are subscribed
+        self.write_and_invalidate(self.plan_2, **{"access_level": "1"})
+        # Remove user_bob from manager  group
+        self.remove_from_group(
+            self.user_bob,
+            [
+                "cetmix_tower_server.group_manager",
+            ],
+        )
+        plan_line_action_read_result = plan_line_action_as_bob.read([])
+        self.assertEqual(
+            plan_line_action_read_result[0]["condition"],
+            plan_line_action_as_bob.condition,
+            msg="Condition should be the same",
+        )
+        # Check user can't access  plan line actions assigned to servers
+        #  to which they aren't subscribed
+        self.server_test_1.message_unsubscribe([self.user_bob.partner_id.id])
+        with self.assertRaises(AccessError):
+            plan_line_action_read_result = plan_line_action_as_bob.read([])
+
+    def test_plan_line_action_variable_values_access_rights(self):
+        # Remove User_bob from all groups
+        self.remove_from_group(
+            self.user_bob,
+            [
+                "cetmix_tower_server.group_user",
+                "cetmix_tower_server.group_manager",
+                "cetmix_tower_server.group_root",
+            ],
+        )
+        # Assign variable value to plan action
+
+        variable_value = self.env["cx.tower.variable.value"].create(
+            {
+                "variable_id": self.variable_os.id,
+                "value_char": "Branch = Main",
+                "plan_line_action_id": self.plan_line_1_action_1.id,
+            }
+        )
+
+        # group_user:
+        # Plan not assigned to server, ensure access to variable value in plan line
+        # action
+        # Add user to group
+        self.add_to_group(self.user_bob, "cetmix_tower_server.group_user")
```

Comment:
You have removed this user from group in line 1582. Why are you adding it back again here? Maybe simply not to remove it at all?

Reply:
Thanks for the comment, 
I removed user_bob in line 1582 from all groups, because this user is used in a large number of tests throughout the module, so I'm not sure if he doesn't belong to any tower.server group at the moment.
I encountered a similar situation a few months earlier.   So I did this just to be sure.
Perhaps there is another, more efficient way to do this check, but I don't know about it.